src/matter/session/secure/SecureChannelStatusMessageSchema.ts

```diff
@@ -0,0 +1,24 @@
+/**
+ * @license
+ * Copyright 2022 The node-matter Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { ByteArray, DataReader, DataWriter, Endian } from "@project-chip/matter.js";
+import { GeneralStatusCode, ProtocolStatusCode, SECURE_CHANNEL_PROTOCOL_ID } from "./SecureChannelMessages";
+
+export function decodeStatusReport(payload: ByteArray) {
```

Comment:
It will be more consistent if you use a real Schema here instead:

```
type StatusMessage {
     generalStatus: GeneralStatusCode,
     protocolId: number,
     protocolStatus: ProtocolStatusCode,
}

export class SecureChannelStatusMessageSchema extends Schema<StatusMessage, ByteArray> {
     encode({ generalStatus, protocolId, protocolStatus }: StatusMessage) {
             const writer = new DataWriter(Endian.Little);
             writer.writeUInt16(generalStatus);
             writer.writeUInt32(protocolId);
             writer.writeUInt16(protocolStatus);
             return writer.toByteArray();
     }
     
     decode(bytes: ByteArray) {
           const reader = new DataReader(bytes, Endian.Little);
          const generalStatus = reader.readUInt16();
          const protocolId = reader.readUInt32();
          const protocolStatus = reader.readUInt16();
          return { generalStatus, protocolId, protocolStatus };
    }
}
```

Reply:
Adjusted